lib/compare.c

```diff
@@ -112,7 +112,9 @@ dx_diff_report_add (struct disir_diff_report *report, const char *fmt, ...)
     va_start (args, fmt);
     do
     {
-        ret = vsnprintf (report->dr_diff_string[i], n, fmt, args);
+        va_list args_copy;
+        va_copy (args_copy, args);
+        ret = vsnprintf (report->dr_diff_string[i], n, fmt, args_copy);
```

Comment:
```suggestion
        ret = vsnprintf (report->dr_diff_string[i], n, fmt, args_copy);
        va_end (args_copy);
```
Each `va_start`/`va_copy` must be matched with a corresponding `va_end` within the same function. See `man stdarg`.